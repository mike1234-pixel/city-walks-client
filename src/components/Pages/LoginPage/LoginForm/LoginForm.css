/** @format */
.login-header-container {
  width: 100%;
  text-align: center;
}

.login-form-header {
  font-family: var(--logo-font);
  margin-top: 30px;
}

/*# sourceMappingURL=LoginForm.css.map */
