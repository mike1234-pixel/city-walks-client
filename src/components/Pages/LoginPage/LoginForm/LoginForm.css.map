{"version":3,"sourceRoot":"","sources":["LoginForm.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA;;;AAGF;EACE;EACA","file":"LoginForm.css"}