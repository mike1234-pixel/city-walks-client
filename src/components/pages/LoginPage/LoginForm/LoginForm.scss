/** @format */

.login-header-container {
  width: 100%;
  text-align: center;
}

.login-form-header {
  font-family: var(--display-font);
}
