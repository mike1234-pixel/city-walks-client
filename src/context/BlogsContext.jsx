import { createContext, useState } from "react"

export const BlogsContext = createContext();

export const BlogsContextProvider = (props) => {

    const [blogPosts, setBlogPosts] = useState([])
    const [blogsLoading, setBlogsLoading] = useState(true)

    return (
        <BlogsContext.Provider 
            value={{
                blogPosts,
                setBlogPosts,
                blogsLoading,
                setBlogsLoading,
            }}
        >
            {props.children}
        </BlogsContext.Provider>
    )
}