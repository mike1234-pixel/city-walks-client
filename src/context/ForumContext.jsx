import { createContext, useState } from "react"
import BoardBox from "../components/pages/LoginPage/LoggedInView/Components/BoardBox/BoardBox"

export const ForumContext = createContext();

export const ForumContextProvider = (props) => {
 
    const [loadingBoards, setLoadingBoards] = useState(true)
    const [boards, setBoards] = useState([])
    const [selectedThreads, setSelectedThreads] = useState([])

    const [currentBoardName, setCurrentBoardName] = useState("")
    const [currentBoardId, setCurrentBoardId] = useState("")

    let displayBoards;

    if (!loadingBoards) {
    displayBoards = boards.map((board, index) => {
        return (
            <BoardBox boardId={board._id} name={board.name} description={board.description} index={index} key={index} />
                )
        })
        // create routes for boards with Thread as the view component
    }

    return (
        <ForumContext.Provider 
            value={{
                setBoards,
                boards, // data
                setLoadingBoards,
                loadingBoards,
                displayBoards,
                setSelectedThreads,
                selectedThreads, 
                setCurrentBoardName,
                currentBoardName,
                setCurrentBoardId,
                currentBoardId,
            }}>
            {props.children}
        </ForumContext.Provider>
    )
}

