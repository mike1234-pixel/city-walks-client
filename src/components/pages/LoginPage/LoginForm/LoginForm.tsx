import React from 'react'
import { useHistory } from "react-router-dom"
import { MDBInput, MDBBtn, MDBIcon, MDBContainer } from "mdbreact"
import UserPortalNav from "../UserPortalNav"
import axios, { AxiosError, AxiosResponse } from "axios"
import qs from "qs"
import { connect } from 'react-redux'
import GlobalState from '../../../../types/State/Global/State'
import Credentials from '../../../../types/PostRequests/Credentials'
import { Dispatch } from 'redux'
import './LoginForm.scss'

interface Props {
  loginEmail: string;
  loginPassword: string;
  setLoginEmail: Function;
  setLoginPassword: Function;
  setLoggedIn: Function;
  setUserId: Function;
  setUserFirstName: Function;
  setUserLastName: Function;
}

const LoginForm: React.FC<Props> = (props: Props) => {

  const { loginEmail, loginPassword, setLoginEmail, setLoginPassword, setLoggedIn, setUserId, setUserFirstName, setUserLastName } = props

  const pushSlug: Function = useHistory().push

  const handleChangeLogin = (e: React.ChangeEvent<any>) => {
    switch (e.target.name) {
      case "login-email":
        setLoginEmail(e.target.value)
        break;
      case "login-password":
        setLoginPassword(e.target.value)
        break;
    }
  }

  const handleSubmitLogin = (e: React.FormEvent) => {
    console.log("handle submit login triggered")
    e.preventDefault()

    const payload: Credentials = {
      email: loginEmail,
      password: loginPassword
    };

    axios
      .post("https://city-walks.herokuapp.com/login-user", qs.stringify(payload))
      .then((res: AxiosResponse) => {
        if (res.data === "Your account exists but is not activated. Please click 'verify account' for email verification.") {
          alert("Your account exists but is not activated. Please click 'verify account' for email verification.")
        } else if (res.data === "unsuccessful login attempt") {
          alert("Unsuccessful Login Attempt. Please Try Again.")
        } else {
          alert("Login Complete")
          setLoggedIn(true)
          setLoginEmail("")
          setLoginPassword("")
          setUserId(res.data._id)
          setUserFirstName(res.data.fname)
          setUserLastName(res.data.lname)
          localStorage.setItem("loggedIn", 'true')
          localStorage.setItem("userId", res.data._id)
          localStorage.setItem("userFirstName", res.data.fname)
          pushSlug("/forum");
          window.scrollTo(0, 0)
        }
      }).catch((err: AxiosError) => {
        console.log(err)
      });
  }

  return (
    <div key="user-login" className="min-page-height">
      <UserPortalNav loggedIn={false} />
      <MDBContainer>
        <div className="login-header-container">
          <h2 className="login-form-header">Login</h2>
          <p>Login to Your Account.</p>
        </div>
        <form onSubmit={handleSubmitLogin} className="user-login-form display-form">
          <MDBInput key="input-5" type="email" name="login-email" id="login-email" value={loginEmail} label="email" onChange={handleChangeLogin} required />
          <MDBInput key="input-6" type="password" name="login-password" id="login-password" value={loginPassword} label="password" onChange={handleChangeLogin} minLength="8" required />
          <MDBBtn outline color="elegant" type="submit">
            Login <MDBIcon icon="sign-in-alt" />
          </MDBBtn>
        </form>
      </MDBContainer>
    </div>
  )
}

const mapStateToProps: (state: GlobalState) => void = (state) => ({
  loginEmail: state.loginState.loginEmail,
  loginPassword: state.loginState.loginPassword,
});

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    setLoginEmail: (loginEmail: string) => dispatch({ type: 'SET_LOGIN_EMAIL', loginEmail }),
    setLoginPassword: (loginPassword: string) => dispatch({ type: 'SET_LOGIN_PASSWORD', loginPassword }),
    setLoggedIn: (boolValue: boolean) => dispatch({ type: 'SET_LOGGED_IN', boolValue }),
    setUserId: (userId: string) => dispatch({ type: 'SET_USER_ID', userId }),
    setUserFirstName: (userFirstName: string) => dispatch({ type: 'SET_USER_ID', userFirstName }),
    setUserLastName: (userLastName: string) => dispatch({ type: 'SET_USER_ID', userLastName }),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginForm);