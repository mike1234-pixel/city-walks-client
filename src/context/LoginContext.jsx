import { createContext, useState } from "react"
import axios from "axios"
import qs from "qs"
import LoginForm from "../components/pages/LoginPage/LoginForm/LoginForm";
import RegistrationForm from "../components/pages/LoginPage/RegistrationForm/RegistrationForm"
import VerificationForm from "../components/pages/LoginPage/VerificationForm/VerificationForm";
import ResetPasswordForm from "../components/pages/LoginPage/ResetPasswordForm/ResetPasswordForm";
import ForgotPasswordForm from "../components/pages/LoginPage/ForgotPasswordForm/ForgotPasswordForm";
import Boards from "../components/pages/LoginPage/LoggedInView/Boards"

export const LoginContext = createContext();

export const LoginContextProvider = (props) => {
    const [popupVisible, setPopupVisible] = useState(false)

    const [firstName, setFirstName] = useState("")
    const [lastName, setLastName] = useState("")
    const [userId, setUserId] = useState("")
    const [registrationEmail, setRegistrationEmail] = useState("")
    const [registrationPassword, setRegistrationPassword] = useState("")
  
    const [loginEmail, setLoginEmail] = useState("")
    const [loginPassword, setLoginPassword] = useState("")
  
    // save all of this in local storage
    const [loggedIn, setLoggedIn] = useState(false)
    const [userFirstName, setUserFirstName] = useState("")
    const [userLastName, setUserLastName] = useState("")
  
    const [verificationEmail, setVerificationEmail] = useState("")

    const [resetPasswordEmail, setResetPasswordEmail] = useState("")
    const [resetPasswordOldPassword, setResetPasswordOldPassword] = useState("")
    const [resetPasswordNewPassword, setResetPasswordNewPassword] = useState("")
    const [resetPasswordConfirmNewPassword, setResetPasswordConfirmNewPassword] = useState("")

    const [forgotPasswordEmail, setForgotPasswordEmail] = useState("")

    const handlePopup = () => {
      setPopupVisible(false)
      localStorage.setItem("popupVisible", false)
    }

    // set display form on LoginPage
    const [form, setForm] = useState("loggedOutView");

    let displayForm = <Boards/>;
    if (form === "boards") {
      displayForm = <Boards/>;
    } else if (form === "loginForm") {
      displayForm = <LoginForm />;
    } else if (form === "registrationForm") {
      displayForm = <RegistrationForm />;
    } else if (form === "verificationForm") {
      displayForm = <VerificationForm />;
    } else if (form === "resetPasswordForm") {
      displayForm = <ResetPasswordForm />;
    } else if (form === "forgotPasswordForm") {
      displayForm = <ForgotPasswordForm />;
    }

    const handleChangeRegistration = (event) => {
        switch(event.target.name) {
            case "registration-fname":
              setFirstName(event.target.value)
              break;
            case "registration-lname":
              setLastName(event.target.value)
              break;
            case "registration-email":
              setRegistrationEmail(event.target.value)
              break;
            case "registration-password":
              setRegistrationPassword(event.target.value)
              break;
          } 
    }
  
    const handleSubmitRegistration = (event) => {
      console.log("handle submit triggered")
        event.preventDefault()
  
        const payload = {
            fname: firstName,
            lname: lastName,
            email: registrationEmail,
            password: registrationPassword,
          };
  
    axios
        .post("http://localhost:5000/register-user", qs.stringify(payload))
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "An account with this email already exists.") {
            alert("An account with this email already exists.")
          } else if (res.data === "We have sent you an email. Please verify your account by clicking the link in the mail.") {
            alert("We have sent you an email. Please verify your account by clicking the link in the email. (This code expires after 10 minutes)")
           } else {
            alert("Registration Complete")
            setLoggedIn(true)
            setFirstName("")
            setLastName("")
            setRegistrationEmail("")
            setRegistrationPassword("")
            setUserFirstName(res.data.fname)
            setUserLastName(res.data.lname)
            window.scrollTo(0, 0)
          }
        });
    }
  
    const handleChangeLogin = (event) => {
      switch(event.target.name) {
          case "login-email":
            setLoginEmail(event.target.value)
            break;
          case "login-password":
            setLoginPassword(event.target.value)
            break;
        } 
  }
  
  const handleSubmitLogin = (event) => {
    console.log("handle submit login triggered")
      event.preventDefault()
  
      const payload = {
          email: loginEmail,
          password: loginPassword
        };
  
  axios
      .post("http://localhost:5000/login-user", qs.stringify(payload))
      .then((res, err) => {
        if (err) {
          console.log(err);
        } else if (res.data === "Your account exists but is not activated. Please click 'verify account' for email verification.") {
          alert("Your account exists but is not activated. Please click 'verify account' for email verification.")
        } else if (res.data === "unsuccessful login attempt") {
          alert("Unsuccessful Login Attempt. Please Try Again.")
        } else {
          alert("Login Complete")
          setLoggedIn(true)
          setLoginEmail("")
          setLoginPassword("")
          setForm("")
          setUserId(res.data._id)
          setUserFirstName(res.data.fname)
          setUserLastName(res.data.lname)
          localStorage.setItem("loggedIn", true)
          localStorage.setItem("userId", res.data._id)
          localStorage.setItem("userFirstName", res.data.fname)
          window.scrollTo(0, 0)
        }
      });
  }

  // resend verification email

  const handleChangeVerification = (event) => {
    switch(event.target.name) {
        case "verification-email":
          setVerificationEmail(event.target.value)
          break;
      }   
  }

const handleSubmitVerification = (event) => {
    console.log("handle submit verification")
    event.preventDefault()

    const payload = {
        email: verificationEmail,
      };

axios
    .post("http://localhost:5000/reverify-user", qs.stringify(payload))
    .then((res, err) => {
      if (err) {
        console.log(err);
      } else {
        alert("Verification email sent. Check your inbox.")
        setVerificationEmail("")
        window.scrollTo(0, 0)
      }
    })
}

  // reset password

  const handleChangeResetPassword = (event) => {
    switch(event.target.name) {
        case "reset-email":
          setResetPasswordEmail(event.target.value)
          break;
        case "old-password":
          setResetPasswordOldPassword(event.target.value)
          break;
        case "new-password":
          setResetPasswordNewPassword(event.target.value)
          break;
        case "confirm-new-password":
          setResetPasswordConfirmNewPassword(event.target.value)
          break;
      }   
  }

const handleSubmitResetPassword = (event) => {
    console.log("handle submit reset password")
    event.preventDefault()

    const payload = {
        email: resetPasswordEmail,
        oldPassword: resetPasswordOldPassword,
        newPassword: resetPasswordNewPassword,
      };

      if (resetPasswordNewPassword === resetPasswordConfirmNewPassword) {
        axios
        .post("http://localhost:5000/reset-password-with-old-password", qs.stringify(payload))
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "user not found") {
            alert("We could not find your account. Please try again.")
          } else if (res.data === "old password does not match password in the database") {
            alert("The password you entered does not match the account. Please try again or use the Forgot Password option.")
          } else if (res.data === "password successfully updated") {
            alert("Password reset. You can now login.")
            setResetPasswordEmail("")
            setResetPasswordOldPassword("")
            setResetPasswordNewPassword("")
            setResetPasswordConfirmNewPassword("")
            window.scrollTo(0, 0)
          }
        })
      } else {
        alert("Passwords don't match. Please try again.")
      }

}

  // forgot password

  const handleChangeForgotPassword = (event) => {
    switch(event.target.name) {
        case "forgot-password-email":
          setForgotPasswordEmail(event.target.value)
          break;
      }   
  }

const handleSubmitForgotPassword = (event) => {
    console.log("handle submit forgot password")
    event.preventDefault()

    const payload = {
        email: forgotPasswordEmail,
      };

axios
    .post("http://localhost:5000/forgot-password", qs.stringify(payload))
    .then((res, err) => {
      if (err) {
        console.log(err);
      } else {
        alert("We have sent you an email. Please click the click in your email to reset your password")
        setForgotPasswordEmail("")
        window.scrollTo(0, 0)
      }
    })
}
  
  const logOut = () => {
    localStorage.clear()
    localStorage.setItem("popupVisible", false)
    setLoggedIn(false)
    setUserId("")
    setUserFirstName("")
    setUserLastName("")
    setForm("boards")
  }

    return (
        <LoginContext.Provider 
            value={{
              popupVisible,
              setPopupVisible,
              handlePopup,
               // display form
               displayForm,
               setForm,
               // logged in state 
               loggedIn,
               // registration state and functions
                firstName,
                lastName,
                registrationEmail,
                registrationPassword,
                handleChangeRegistration,
                handleSubmitRegistration,
                // login state and functions
                loginEmail,
                loginPassword,
                handleChangeLogin: handleChangeLogin,
                handleSubmitLogin: handleSubmitLogin,
                setLoggedIn,
                setUserFirstName,
                setUserId,
                // logged in user data
                userFirstName,
                userId,
                userLastName,
                // resend verification email
                verificationEmail,
                handleChangeVerification,
                handleSubmitVerification,
                // reset password
                resetPasswordEmail,
                resetPasswordOldPassword,
                resetPasswordNewPassword,
                resetPasswordConfirmNewPassword,
                handleChangeResetPassword,
                handleSubmitResetPassword,
                // forgot password
                forgotPasswordEmail,
                handleChangeForgotPassword,
                handleSubmitForgotPassword,
                // logout function
                logOut
            }}
        >
            {props.children}
        </LoginContext.Provider>
    )
}

// // setter
// localStorage.setItem('myData', data);
 
// // getter
// localStorage.getItem('myData');
 
// // remove
// localStorage.removeItem('myData');
 
// // remove all
// localStorage.clear();

