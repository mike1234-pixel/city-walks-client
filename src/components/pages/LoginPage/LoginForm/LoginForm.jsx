import { useContext } from 'react'
import { LoginContext } from '../../../../context/LoginContext'
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact"
import './LoginForm.css'

const LoginForm = () => {

      const {
        loginEmail, 
        loginPassword, 
        handleChangeLogin, 
        handleSubmitLogin} = useContext(LoginContext)

    return (
    <div key="user-login">
      <div className="login-header-container">
        <h2 className="login-form-header">Login</h2>
        <p>Login to Your Account.</p>
      </div>
        <form onSubmit={handleSubmitLogin} className="user-login-form display-form">
            <MDBInput key="input-5" type="email" name="login-email" id="login-email" value={loginEmail} label="email" onChange={handleChangeLogin} required/>
            <MDBInput key="input-6" type="password" name="login-password" id="login-password" value={loginPassword} label="password" onChange={handleChangeLogin} minLength="8" required/>
            <MDBBtn outline color="elegant" type="submit">
                Login <MDBIcon icon="sign-in-alt" />
          </MDBBtn>
        </form>
    </div>
    )
}

export default LoginForm