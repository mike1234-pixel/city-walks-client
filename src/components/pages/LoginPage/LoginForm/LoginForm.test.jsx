import renderer from 'react-test-renderer'
import { LoginContextProvider } from "../../../../context/LoginContext"
import LoginForm from "./LoginForm"

// snapshot test
it('LoginForm renders correctly', () => {  

  const tree = renderer
    .create(
            <LoginContextProvider>
                <LoginForm />
            </LoginContextProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});