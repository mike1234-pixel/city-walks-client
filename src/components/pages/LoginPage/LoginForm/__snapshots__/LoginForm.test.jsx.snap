// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
<div>
  <div
    className="login-header-container"
  >
    <h2
      className="login-form-header"
    >
      Login
    </h2>
    <p>
      Login to Your Account.
    </p>
  </div>
  <form
    className="user-login-form display-form"
    onSubmit={[Function]}
  >
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="login-email"
        name="login-email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="email"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="login-email"
        id=""
        onClick={[Function]}
      >
        email
      </label>
    </div>
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="login-password"
        minLength="8"
        name="login-password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="password"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="login-password"
        id=""
        onClick={[Function]}
      >
        password
      </label>
    </div>
    <button
      className="btn-elegant btn-outline-elegant btn Ripple-parent"
      data-test="button"
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="submit"
    >
      Login 
      <i
        className="fa fa-sign-in-alt"
        data-test="fa"
      />
      <div
        className="Ripple Ripple-outline "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </form>
</div>
`;
